utils.js

```diff
@@ -0,0 +1,61 @@
+
+function getDistanceFromLatLonInKm(lat1,lon1,lat2,lon2) {
+    var R = 6371; // Radius of the earth in km
+    var dLat = deg2rad(lat2-lat1);  // deg2rad below
+    var dLon = deg2rad(lon2-lon1); 
+    var a = 
+        Math.sin(dLat/2) * Math.sin(dLat/2) +
+        Math.cos(deg2rad(lat1)) * Math.cos(deg2rad(lat2)) * 
+        Math.sin(dLon/2) * Math.sin(dLon/2)
+        ; 
+    var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a)); 
+    var d = R * c; // Distance in km
+    return d;
+}
+
+function deg2rad(deg) {
+    return deg * (Math.PI/180)
+}
+
+function calculateRiskScore(distance, timeElapsedInHours) {
+    const strictScale = 1;
+    const airTravelSpeed = 800; // km/hour
+    const groundTravelSpeed = 120; // km/hour
+    const distanceBuffer = 250;
+    const adjustedgroundTravelSpeed = groundTravelSpeed + distanceBuffer/timeElapsedInHours; // km/hour
+    const adjustedairTravelSpeed = airTravelSpeed + distanceBuffer/timeElapsedInHours; // km/hour
+    
+    const adjustedSpeed = (distance + distanceBuffer)/timeElapsedInHours ;
+
+    // Base speed is 0.5 * 
+
+    // Distance = 0, Time = 0.1, Base speed = 0 + 250/0.1 = 500
+    // Distance = 0, Time = 8, Base speed = 0 + 250/8 = 31.25
+
```

Comment:
what happens if someone is using the application in flight and the IP changes accordingly?

Reply:
Added change to update the subsequent IP addresses in DB. Had planned initially to do it but deprioritised it for getting the service up and running.